telemetry/config.go

```diff
@@ -0,0 +1,84 @@
+// Package telemetry provides instrumentation tools for other modules and packages.
+package telemetry
+
+import (
+	"fmt"
+	"time"
+
+	gometrics "github.com/armon/go-metrics"
+	"github.com/spf13/viper"
+)
+
+// Config defines the configuration options for application telemetry.
+type Config struct {
+	// Prefixed with keys to separate services
+	ServiceName string `mapstructure:"service-name"`
+
+	// Enabled enables the application telemetry functionality. When enabled,
+	// an in-memory sink is also enabled by default. Operators may also enabled
+	// other sinks such as Prometheus.
+	Enabled bool `mapstructure:"enabled"`
+
+	// Enable prefixing gauge values with hostname
+	EnableHostname bool `mapstructure:"enable-hostname"`
+
+	// Enable adding hostname to labels
+	EnableHostnameLabel bool `mapstructure:"enable-hostname-label"`
+
+	// Enable adding service to labels
+	EnableServiceLabel bool `mapstructure:"enable-service-label"`
+
+	// PrometheusRetentionTime, when positive, enables a Prometheus metrics sink.
+	// It defines the retention duration in seconds.
+	PrometheusRetentionTime time.Duration `mapstructure:"prometheus-retention-time"`
+
+	// GlobalLabels defines a global set of name/value label tuples applied to all
+	// metrics emitted using the wrapper functions defined in telemetry package.
+	//
+	// Example:
+	// [["chain_id", "cosmoshub-1"]]
+	GlobalLabels [][]string `mapstructure:"global-labels"`
+
+	//Suggestion : use the gometrics.Lable and update the config contract. also provide proper Options for best accessibility.
+	globalLabels []gometrics.Label
+
+	useGlobalMetricRegistration bool
```

Comment:
Can we add a godoc for this? Describe what it does and what true/false means please 🙏

Reply:
resolved